test: add visual regression tests

LondonOktober opened the pull request:
Greetings!

Since bootstrap (and by extension, bootstrap-vue) contains a preset theme of styles for styling apps I figured Visual Testing, where snapshots are taken and compared by the pixel, would be a beneficial regression test for this awesome library.

I've added Cypress ([cypress.io](https://cypress.io) - Testing harness) and Applitools ([applitools.com](https://applitools.com) - Visual regression checker) to the documentation examples. Applitool's SDK takes DOM snapshots w/ styles (not screenshots) from headless Chrome. This will help alert if there are any unintended changes to the rendered examples of the documentation.

<img width="1437" alt="Screen Shot 2019-05-28 at 7 18 50 PM" src="https://user-images.githubusercontent.com/10950960/58524901-a116d000-817e-11e9-8231-331dcaf370af.png">

I hope you find this useful! 

Beyond what I have included, you will need to sign up for a (free for open source) account at Applitools to get an account API Key. You will probably also want to set it up to run on pull requests.

tmorehouse commented on package.json:
```diff
@@ -69,6 +69,7 @@
     "postinstall": "opencollective || exit 0"
   },
   "dependencies": {
+    "@applitools/eyes-cypress": "^3.6.15",
```
I'm not sure this should be in `dependencies`, but rather should be `devDependencies`, since it is not required to install BootstrapVue into a project.

LondonOktober replied:
My mistake. Fixed. Thank you!

LondonOktober commented:
@tmorehouse @pi0 

I have rebased against `dev` to resolve conflicts and addressed current feedback. I am unsure how to proceed with the very slight drop in code coverage as I have not modified existing code beyond package/yarn files.

Please advise. Thank you!

pi0 commented:
Hi @LondonOktober. Thanks for updates. Covdecov report is wrong we can just ignore it for now. Would you please ping me on [discord](https://discord.gg/j2Mtcny) chat? Would nice having a room for more discussion.

cmwhited commented:
@LondonOktober looks like there are still a couple conflicting files. Can we take a look and @pi0, @tmorehouse work on getting this merged in?

LondonOktober commented:
@pi0 @tmorehouse Updated with no conflicts, ready for re-review. Thank you!

cmwhited commented:
@LondonOktober I am still seeing conflicts in the `package.json` and `yarn.lock` files. @pi0 @tmorehouse any update with this? are we good to merge once those have been resolved?

cmwhited commented:
Pinging again: @LondonOktober I am still seeing conflicts in the package.json and yarn.lock files. @pi0 @tmorehouse any update with this? are we good to merge once those have been resolved?

LondonOktober commented:
@pi0 @tmorehouse @jackmu95  @cmwhited Updated with no conflicts, ready for re-review. Thank you!

cmwhited commented:
Thanks @LondonOktober! @pi0 @tmorehouse can y'all take a look at this so we can hopefully get it merged?

cmwhited commented:
@jackmu95 if you wanted to resolve those conflicts, that would be much appreciated.

As for the benefit, Applitools provides an awesome service for automated, AI-driven, visual regression testing. We believe that the repo would benefit from this tech.

cmwhited commented:
@jackmu95 did my comment on the benefits make sense? If so, can we get this merged?

cmwhited commented:
@jackmu95 pinging on this again. Would love to get some eyes on it and get it merged

jackmu95 commented:
@cmwhited My opinion hasn't changed.
I leave the decision wether we merge this up to @pi0 and @tmorehouse.

cmwhited commented:
@jackmu95 thanks. @pi0 @tmorehouse can y'all take a look at let us know what you think?

cmwhited commented:
@pi0 @tmorehouse have you had a chance to take a look at this PR? would love to get it merged. thanks!

cmwhited commented:
@pi0 @tmorehouse @jackmu95 pinging again. thanks

cmwhited commented:
@tmorehouse is there anything we can do to get eyes on this PR? would really love to get it merged. Let me know

cmwhited commented:
@tmorehouse pinging again. thanks

cmwhited commented:
@pi0 glad to see this has been reopened. Let me know if we can help with anything!

pi0 commented:
@cmwhited Also glad you are still in this improvement! I hope @LondonOktober you are not disappointed after this long delay. This is how OSS projects sometimes go 🙈I think we need a plan to revamp visual-tests. I will calendar a quick call with you if it's ok for next week.

cmwhited commented:
@pi0 that works for me. let me know.
